include/wx/osx/dataobj.h

```diff
@@ -11,28 +11,31 @@
 #ifndef _WX_MAC_DATAOBJ_H_
 #define _WX_MAC_DATAOBJ_H_
 
-// ----------------------------------------------------------------------------
-// wxDataObject is the same as wxDataObjectBase under wxGTK
-// ----------------------------------------------------------------------------
+class WXDLLIMPEXP_CORE wxOSXDataSink;
+class WXDLLIMPEXP_CORE wxOSXDataSource;
 
 class WXDLLIMPEXP_CORE wxDataObject : public wxDataObjectBase
 {
 public:
     wxDataObject();
-#ifdef __DARWIN__
     virtual ~wxDataObject() { }
-#endif
 
     virtual bool IsSupportedFormat( const wxDataFormat& format, Direction dir = Get ) const;
+
+    void Write(wxOSXDataSink *sink) const;
```

Comment:
Those should really be prefixed with `OSX` to prevent conflicts with methods with the same names in user-defined subclasses.

Reply:
I'd like to keep the method names OSX free if possible, I thought of

```
    void WriteToSink(wxOSXDataSink *sink) const;
    bool ReadFromSource(wxOSXDataSource *source);
    bool CanReadFromSource(wxOSXDataSource *source) const;

```

what do you think ?

Reply:
IMO it doesn't really make much sense to have public method names when they operate with private classes. What are we going to do if/when we replace `wxOSXDataSink` with `wxDataSink` later?

BTW, I wonder if they actually have to be members of `wxDataObject` at all, AFAICS they could be implemented as free functions (which could be declared in OS X private header and so called anything you want) too and this might be preferable until we agree on the public API changes.

Reply:
I could already change the class names to `wxDataSink` and `wxDataSource`, also make the methods virtual, allowing subclasses of `wxDataObject` to apply their data in a specialized way...

Reply:
This risks being much more time-consuming than "just" fixing macOS clipboard support. E.g. we'd need documentation (adding the docs at some later, unspecified moment of time just doesn't work, so it would really need to be done now), ideally some example of using them in the sample and (a man can dream, right?) maybe some unit tests.

I really wouldn't want to delay fixing the problem(s?) with the clipboard under macOS 10.14 until all this is done, it doesn't seem worth it to me... but maybe I'm missing the importance of these changes, of course.

Reply:
If I put the classes into a private header, and but still rename them to wxData... and only have a forward decl in the public header, would this still be ok as an intermediate step for you ? I'd like to test the API against iOS d&d in the next step, but after a merge. And in 10.14.4 the bug will be fixed by Apple in the OS.

Reply:
naming the include to `osx/private/datatransfer.h` ?